### ISSUE TITLE
Remove twisted/python/_epoll.c

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5847]
|[<img alt="vperic's avatar" src="https://avatars.githubusercontent.com/u/688281?s=50" width="50" height="50">](https://github.com/vperic)| @vperic reported|
|-|-|
|Trac ID|trac#5847|
|Type|task|
|Created|2012-08-04 10:55:09Z|
|Branch|https://github.com/twisted/twisted/tree/remove-epoll-5847-2|

Our own epoll wrapper is not built on Python2.6+. As older Python versions are no longer supported, we can remove the module and instead use the implementation in the official library.

Attched patch removes mentions of it in twisted/topfiles/setup.py and a related private helper from twisted.python.dist. The _epoll module itself (t.p._epoll.{c,pyx}) and it's tests (t.test.test_epoll) have also been removed. Finally, t.internet.epollreactor has been changed to directly refer to the select module (rather than calling it _epoll) and a conditional import was removed.

Attachments:

* [remove-epoll.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/fb6/fb6c0ab1d76ffb39e0c99ad301182e70b326b3d1/44a39050cb4c9cf1e41903ff7d9deb17a8b9048a.patch) (160190 bytes) - added by vperic on 2012-08-04 10:55:32Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5847 5847
type__task task
reporter__vperic vperic
priority__normal normal
milestone__ 
branch__branches_remove_epoll_5847_2 branches/remove-epoll-5847-2
branch_author__therve__vperic therve, vperic
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1344077709000000 1344077709000000
changetime__1362594766000000 1362594766000000
version__None None
owner__therve therve
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst set owner to @vperic|
|-|-|

Looks good; merge with a news file (a .misc is fine, since this change is not user visible in any way), and as usual only if buildbot passes on all platforms.

**Comment by twisted-trac** [comment]
|[<img alt="vperic's avatar" src="https://avatars.githubusercontent.com/u/688281?s=50" width="50" height="50">](https://github.com/vperic)<a name="note_2"></a>|@vperic commented|
|-|-|

(In [35187]) Create branch remove-_epoll-5847

**Comment by twisted-trac** [comment]
|[<img alt="vperic's avatar" src="https://avatars.githubusercontent.com/u/688281?s=50" width="50" height="50">](https://github.com/vperic)<a name="note_3"></a>|@vperic commented|
|-|-|

[Build results.](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/remove-_epoll-5847)

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst commented|
|-|-|

There's test failure on FreeBSD and some of the other non-Linux platforms.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_7"></a>|@therve commented|
|-|-|

(In [37427]) Branching to 'remove-epoll-5847-2'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_8"></a>|@therve removed owner|
|-|-|

This is back for review, builds here: http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/remove-epoll-5847-2

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set owner to @therve<br>@exarkun set status to `new`|
|-|-|

Thanks.  I'm slightly concerned that the stdlib-based epoll implementation is broken under load (<http://twistedmatrix.com/pipermail/twisted-python/2013-March/026622.html>) but I guess it's easy to revert this change later if it turns out we want to restore our version of the bindings for some reason.

Apart from that, the builds look good.  The news file is missing, though.  Please fix that and merge.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_11"></a>|@therve set status to `closed`|
|-|-|

(In [37439]) Merge remove-epoll-5847-2

Authors: vperic, therve
Reviewers: exarkun, itamar
Fixes: [#5847](https://github.com/twisted/twisted/issues/5847)

Remove custom epoll bindings, the epoll reactor using stdlib version since
python 2.6.
